extensions/pvaPublish/webpack.config.js

```diff
@@ -1,54 +1,66 @@
 const { resolve } = require('path');
 const svgToMiniDataURI = require('mini-svg-data-uri');
 
-module.exports = [
-  {
-    entry: './src/ui/index.tsx',
-    mode: 'production',
-    output: {
-      filename: 'publish-bundle.js',
-      path: resolve(__dirname, 'lib', 'ui'),
-    },
-    externals: {
-      // expect react & react-dom to be available in the extension host iframe
-      react: 'React',
-      'react-dom': 'ReactDOM',
-    },
-    module: {
-      rules: [
-        {
-          test: /\.tsx?$/,
-          loader: 'ts-loader',
-        },
-        {
-          test: /\.svg$/i,
-          loader: 'url-loader',
-          options: {
-            generator: (content) => svgToMiniDataURI(content.toString()),
+function getWebpackConfigs() {
+  const isDevelopment = process.env.NODE_ENV === 'development';
+  console.log(`[pvaPublish] Generating ${isDevelopment ? 'development' : 'production'} bundles.`);
+
+  return [
+    {
+      entry: './src/ui/index.tsx',
+      mode: isDevelopment ? 'development' : 'production',
+      devtool: isDevelopment ? 'source-map' : undefined,
+      output: {
+        filename: 'publish-bundle.js',
+        path: resolve(__dirname, 'lib', 'ui'),
+      },
+      externals: {
+        // expect react & react-dom to be available in the extension host iframe
+        react: 'React',
+        'react-dom': 'ReactDOM',
+      },
+      module: {
+        rules: [
+          {
+            test: /\.tsx?$/,
+            loader: 'ts-loader',
           },
-        },
-      ],
-    },
-    resolve: {
-      extensions: ['.js', '.ts', '.tsx'],
-    },
-    plugins: [],
-  },
-  {
-    entry: './src/node/index.ts',
-    mode: 'production',
-    devtool: 'source-map',
-    target: 'node',
-    output: {
-      path: resolve(__dirname, 'lib', 'node'),
-      filename: 'index.js',
-      libraryTarget: 'commonjs2',
-    },
-    module: {
-      rules: [{ test: /\.tsx?$/, use: 'ts-loader', exclude: [/node_modules/] }],
+          {
+            test: /\.svg$/i,
+            loader: 'url-loader',
+            options: {
+              generator: (content) => svgToMiniDataURI(content.toString()),
+            },
+          },
+        ],
+      },
+      resolve: {
+        extensions: ['.js', '.ts', '.tsx'],
+      },
+      plugins: [],
     },
-    resolve: {
-      extensions: ['.js', '.ts', '.tsx'],
+    {
+      entry: './src/node/index.ts',
+      mode: isDevelopment ? 'development' : 'production',
+      target: 'node',
+      devtool: isDevelopment ? 'source-map' : undefined,
```

Comment:
shouldn't this be the other way around?

Reply:
I want it to behave like this:

```
if (isDevelopment)
  // generate source maps
else 
  // don't generate source maps
```

Reply:
interesting. it's usually the other way around. until we start reporting errors, I guess it doesn't matter too much.

Reply:
I would think you would want source maps when developing so you can step through your code and work out any bugs, and then you would not want them in production to reduce size and because you aren't often debugging the production code.

We also don't copy the source files of the extensions into the production app so the source maps wouldn't even work would they?

Reply:
Normally, you would use something like `cheap-eval-source-map` for dev because of build speed. In this case, we aren't concerned as much with build speed.